src/sql/workbench/contrib/connection/common/connectionProviderExtension.ts

```diff
@@ -132,6 +132,20 @@ const ConnectionProviderContrib: IJSONSchema = {
 					defaultValue: {
 						type: 'any'
 					},
+					defaultValueOsOverrides: {
+						type: 'array',
+						items: {
+							type: 'object',
+							properties: {
+								osEnumValue: {
+									type: 'integer'
```

Comment:
I would make this an enum type, with the values being 'darwin', 'win32', 'linux'. and rename the field to just os.

Reply:
Let me do that.

Reply:
Done.